wandb/sdk/lib/settings_file.py

```diff
@@ -0,0 +1,196 @@
+"""Reading and writing W&B settings files.
+
+There is usually a "global" settings file at ~/.config/wandb/settings.
+A "local" settings file, also referred to as "workspace settings", is created
+by the `wandb init` CLI command at ./wandb/settings. Any settings defined in
+the local file override settings in the global file.
+
+These settings files are generally updated by CLI commands like `wandb online`,
+`wandb offline` and `wandb login`, but it is possible to modify them manually.
+The most common settings found in these files are `base_url` (the W&B server
+URL if not using SaaS) and `mode` (online or offline).
+"""
+
+from __future__ import annotations
+
+import configparser
+import logging
+import pathlib
+
+from wandb.errors import Error, term
+
+# The only section name we recognize in all settings files.
+_SECTION_NAME = "default"
+
+
+_logger = logging.getLogger(__name__)
+
+
+class SaveSettingsError(Error):
+    """Could not update one or more settings files.
+
+    The __str__ representation is written to be printed to a terminal.
+    """
+
+
+class SystemSettings:
```

Comment:
In my mind, this implies something like "os-level settings". Maybe FileBackedSettings? Or just Settings since you're using this as settings_file.<Class name>?

Reply:
don't draw inspiration from there, haha. main thing, it's about these coming from files somewhere on disk, which is communicated by the settings_file. part.